packages/modal/src/ux-modal.ts

```diff
@@ -0,0 +1,344 @@
+    }
+    this.showing = true;
+    this.modalService.addLayer(this, this.bindingContext);
+    this.setZindex();
+    // We rely on `queueTask()` here to make sure the 
+    // element is completely ready with all CSS set
+    // before to set `showed = true` which will start
+    // the CSS transition to bring the modal to the
+    // screen
+    this.taskQueue.queueTask(() => {
+      this.showed = true;
+    });
+  }
+
+  private async hide() {
+    if (this.hiding || !this.showed) {return;}
+    this.hiding = true;
+    const duration = this.getAnimationDuration();
+    this.showed = false;
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        this.modalService.removeLayer(this);
+        this.hiding = false;
+        if (this.lastActiveElement && typeof this.restoreFocus === 'function') {
+          this.restoreFocus(this.lastActiveElement);
+        }
+        resolve();
+      }, duration);
+    });
+  }
+
+  private setZindex() {
+    if (this.overlayElement) {
+      this.overlayElement.style.zIndex = `${this.modalService.zIndex}`;
+    }
+    this.contentWrapperElement.style.zIndex = `${this.modalService.zIndex}`;
+  }
+
+  private moveToHost() {
+    const host = this.getHost();
+    if (!host) {
+      return
+    };
+    host.appendChild(this.element);
+  }
+
+  private removeFromHost() {
+    const host = this.getHost();
+    if (!host) {
+      return
+    };
+    try {
+      host.removeChild(this.element);
+    } catch (e) {
+      // if error, it's because the child is already removed
+    }
+  }
+
+  private getHost(): Element | undefined {
+    if (this.host === 'body') {
+      return document.body;
+    } else if (this.host instanceof HTMLElement) {
+      return this.host;
+    } else if (typeof this.host === 'string') {
+      return document.querySelector(this.host) || undefined;
+    }
+    return undefined;
+  }
+
+  public unbind() {
+    window.removeEventListener('resize', this);
+  }
+
+  public handleEvent() {
+    if (this.handlingEvent) {
+      return;
+    }
+    this.handlingEvent = true;
+    if (PLATFORM.global.requestAnimationFrame) {
+      PLATFORM.global.requestAnimationFrame(() => {
+        this.setViewportType();
+        this.handlingEvent = false;
+      });
+    } else {
+      setTimeout(() => {
+        this.setViewportType();
+        this.handlingEvent = false;
+      }, 100);
+    }
+    this.setViewportType();
+  }
+
+  public themeChanged(newValue: any) {
+    if (newValue != null && newValue.themeKey == null) {
+      newValue.themeKey = 'modal';
+    }
+
+    this.styleEngine.applyTheme(newValue, this.element);
+  }
+
+  public setViewportType() {
+    this.viewportType = window.innerWidth < this.modalBreakpoint ? 'mobile' : 'desktop';
+  }
+
+  @computedFrom('type', 'viewportType')
+  public get computedType(): 'standard' | 'modal' {
+    return this.viewportType === 'mobile' ? 'modal' : this.type;
```

Comment:
Do we need to normalize `this.type` return here to either `standard` or `modal`?